packages/web-app-files/src/components/SideBar/Shares/Collaborators/RecipientContainer.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <oc-recipient class="files-share-invite-recipient" :recipient="formattedRecipient">
+    <template #append>
+      <oc-button
+        class="files-share-invite-recipient-btn-remove"
+        appearance="raw"
+        :aria-label="btnDeselectRecipientLabel"
+        @click.stop="deselect(recipient)"
+      >
+        <oc-icon name="close" size="small" />
+      </oc-button>
+    </template>
+  </oc-recipient>
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+
+import { shareTypes } from '../../../../helpers/shareTypes'
+
+export default {
+  props: {
+    recipient: {
+      type: Object,
+      required: true
+    },
+    deselect: {
+      type: Function,
+      required: false,
+      default: null
+    }
+  },
+
+  data() {
+    return {
+      formattedRecipient: {
+        name: this.recipient.label,
+        icon: this.getRecipientIcon(),
+        hasAvatar: this.recipient.value.shareType === shareTypes.user,
+        isLoadingAvatar: true
+      }
+    }
+  },
+
+  computed: {
+    ...mapGetters(['configuration']),
+
+    btnDeselectRecipientLabel() {
+      const translated = this.$gettext('Deselect %{name}')
+
+      return this.$gettextInterpolate(translated, { name: this.recipient.label })
+    }
+  },
+
+  created() {
+    if (this.recipient.value.shareType === shareTypes.user) {
+      this.fetchRecipientAvatar()
+    }
+  },
+
+  methods: {
+    async fetchRecipientAvatar() {
+      try {
```

Comment:
We have an avatar capability now. Please add an early return if it's falsy. See https://github.com/owncloud/web/blob/3f2db716e5a2d333b4714fc2a19ab6f333904dcd/packages/web-runtime/src/components/Avatar.vue#L85 for reference. Would be nice to extract the avatar fetching logic into a helper (since it's used here and in the `Avatar.vue` component of the runtime), but not strictly necessary in this PR.

Reply:
Huh, thanks @fschade for the pointer to the avatar fetching helper, didn't spot it yesterday. Why is our own avatar component not using it though? 🤔